cmd/kubermatic-api/swagger.json

```diff
@@ -17886,7 +17886,7 @@
           "x-go-name": "BucketName"
         },
         "credentials": {
-          "$ref": "#/definitions/ObjectReference"
+          "$ref": "#/definitions/SecretReference"
```

Comment:
@zreigz is our API ready to handle such change?

Reply:
The CRD changes with backup destinations got merged last week(https://github.com/kubermatic/kubermatic/pull/8021)
The API changes with above got merged today. (https://github.com/kubermatic/kubermatic/pull/8228)
Dashboard is not using this yet, and nothing is using this in the API, and the CRD changes are not used anywhere yet as well. 
So its a good moment to change this.

Reply:
:+1: